Vero destination docs (#719)
rudderlabs/rudderstack-docs: docs/destinations/marketing/vero/cloud-mode.mdx

---
title: "Cloud Mode"
description: >-
  Detailed technical documentation on sending events to Vero via the RudderStack cloud mode.
---

RudderStack lets you send your event data to Vero via the <Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud mode</Link>.

<div class="infoBlock">

> Find the open source transformer code for this destination in the <a href="">GitHub repository</a>.

Author: @yashasvibajpai would need the link to the transformer code once it's ready/live.

</div>

## Identify

The <Link to="/event-spec/standard-events/identify">`identify`</Link> call lets you identify a user and associate them to their actions. It also lets you record any traits about them like their name, email, etc.

RudderStack sends the `identify` events to Vero via their [`users`](https://api.getvero.com/api/v2/users/track) endpoint to create, update, and manage your user records and their subscription status.

A sample `identify` call is shown below:

```javascript
rudderanalytics.identify("1hKOmRA4GRlm", {
  "email": "alex@example.com",
  "gender": "Male",
  "randomKey": "randomData"
})
```

### Supported mappings

The following table shows the mapping of the RudderStack properties to the corresponding Vero fields:

| RudderStack trait | Vero property | Presence |
| :-----| :------| :-------|
| `userId` | `id` | Required

> | `email` | `email` | Conditionally required |

Author: Here, we mean that Vero uses `email` as the `id` if `id` is not present, right? Have added the relevant note below, but just want to confirm if that's what we mean by "Conditionally required".

Reviewer: Yes exactly. Though I am not sure if `conditionally required` is a suitable word for it. If you have any better alternative we can go with that too.

</div>

<div class="successBlock">
You can add or update user records that you want as these properties are completely custom.
</div>

## Track

The <Link to="/event-spec/standard-events/track">`track`</Link> call lets you capture user events along with the properties associated with them.

RudderStack sends the `track` events to Vero via their [`events`](https://api.getvero.com/api/v2/events/track) endpoint to track events for a specific user based on the actions they take. If a user profile does not exist, Vero will create one.

A sample `track` event is as shown:

```javascript
rudderanalytics.track("Product Viewed", {
  SKU: "P001",
  "revenue": 77.6,
  "currency": "USD",
  "review_id": "R1619" 
})
```

<div class="infoBlock">
Vero's API treats the capitalized or lower case letters, spaces, and underscores in the event names in the same way. For example, Vero will match <code class="inline-code">Purchased Item</code>, <code class="inline-code">purchased item</code>, and <code class="inline-code">purchased_item</code> as the same event.

> </div>

Author: @yashasvibajpai Is this the case for web device mode too? I saw the note here but not in that section, so confirming.

Reviewer: Yes, this is a uniform behaviour to both modes

### Supported mappings

The following table shows the mapping of the RudderStack properties to the corresponding Vero fields:

| RudderStack property | Vero property | Presence |
| :-----| :------| :-------|
| `userId`/`anonymousId` | `identity.id` | Required |
| `event`  | `event_name` | Required |
| `email` | `identity.email` | Optional |
| `properties` | `data` | Optional |

## Alias

RudderStack sends the <Link to="/event-spec/standard-events/alias">`alias`</Link> events to Vero via their [`users`](https://api.getvero.com/api/v2/users/reidentify) endpoint to change a user's identifier (`id`).

A sample `alias` call is shown below:

```javascript
rudderanalytics.alias("userId", "previousId");
```

### Supported mappings

The following table shows the mapping of the RudderStack properties to the corresponding Vero fields:

| RudderStack property | Vero property | Presence |
| :-----| :------| :-------|
| `userId` | `new_id` | Required |
| `previousId` | `id` | Required |

> ## Adding and removing tags

Author: @yashasvibajpai Is this feature supported only for `identify`? or is `track` supported as well? Please confirm, so that I can move this section accordingly.

Reviewer: This is supported for both `Identify` and `Track` by passing via `integrations` object.

> RudderStack supports adding and removing `tags` to all the `identify` and `track` calls by passing them via the `integrations` object. An example snippet is shown below:

Reviewer: ```suggestion
RudderStack supports adding and removing `tags` to `identify`, `track`, `page` and `screen` calls by passing them via the `integrations` object. An example snippet is shown below:
```
CC @ItsSudip

```javascript
rudderanalytics.identify(
  "1hKOmRA4GRlm", {
    email: "alex@example.com"
  }, {
    integrations: {
      Vero: {
        tags: {
          add: ["alex", "keener", "neworleans"],
          remove: ["james", "ducker", "newyork"]
        }
      }
    }
  }
);
```

> 

Author: @yashasvibajpai Once the tags are added or removed, how does it impact the fields in Vero? Can we add some example/reference here?

Reviewer: Adding or removing tags attaches these tags to the appropriate user. We can see them in the user's profile in Vero dashboard. Should we have a screenshot here for same?

Author: I think it will be very helpful. could you please share an example image?

---

RudderStack lets you send your event data to Vero via the <Link to="/rudderstack-cloud/rudderstack-connection-modes/#cloud-mode">cloud mode</Link>.

<div class="infoBlock">
Find the open source transformer code for this destination in the <a href="">GitHub repository</a>.
</div>

## Identify

The <Link to="/rudderstack-api/api-specification/rudderstack-spec/identify/">`identify`</Link> call lets you identify a user and associate them to their actions. It also lets you record any traits about them like their name, email, etc.

RudderStack sends the `identify` events to Vero via their [`users`](https://developers.getvero.com/?bash#users) endpoint to create, update, and manage your user records and their subscription status.

A sample `identify` call is shown below:

```javascript
rudderanalytics.identify("1hKOmRA4GRlm", {
  "email": "alex@example.com",
  "gender": "Male",
  "randomKey": "randomData"
})
```

### Supported mappings

The following table lists the mapping of the RudderStack traits to the corresponding Vero properties:

| RudderStack trait | Vero property | Presence |
| :-----| :------| :-------|
| `userId` | `id` | Required, if `email` is not present. |

> | `email` | `email` | Required, if `id` is not present. |

Author: ```suggestion
| `email` | `email` | Required, if `userId` is not present. |
```

| `traits` | `data` |  Optional |
| `context.os.name` | `channels.platform` | Optional |
| `context.device.token` | `channels.address` | Optional |
| `context.device.name` | `channels.device` | Optional |
| `context.timezone` | `timezone` | Optional |
| `context.locale`| `locale` | Optional |
| `context.userAgent` | `userAgent` | Optional |

<div class="infoBlock">

> If the event includes <code class="inline-code">email</code> and no <code class="inline-code">id</code>, Vero will automatically use <code class="inline-code">email</code> as the <code class="inline-code">id</code>.

Author: ```suggestion
If the event includes <code class="inline-code">email</code> and no <code class="inline-code">userId</code>, Vero will automatically use <code class="inline-code">email</code> as the <code class="inline-code">id</code>.
```
